### PR TITLE
BK-4455: Added a keepalive feature

### PR DESCRIPTION
**Pull request by shoham-stratoscale**
If the keepalive timeout has passed, renew the session, before the server does

@Stratoscale/mgmt @eyal-stratoscale @roeih-stratoscale @adir-stratoscale

### CODE REVIEW

File: hammock/templates/client/file.j2 [hunk 1]
```diff
@@ -140,7 +146,9 @@ class {{ class_name }}(object):
             _kwargs.setdefault('headers', {}).update(headers)
         LOG.debug("Sending %s request: %s %s", method, url, json)
         try:
+            self._renew_keepalive()
             response = getattr(self._client, method.lower())(url, **_kwargs)
+            self._last_response = time.time()
```
Comment (maroun-stratoscale) [inline hammock/templates/client/file.j2]: Maybe it's better to name it `_last_response_time`

File: hammock/templates/client/file.j2 [hunk 2]
```diff
@@ -164,6 +172,15 @@ class {{ class_name }}(object):
...
         return self._client.headers.get("{{ token_entry }}")
 
+    def _renew_keepalive(self):
+        if self._last_response is None:
+            return
+        time_since_last_response = time.time() - self._last_response
+        if time_since_last_response > self._keepalive_timeout:
+            # Close the session, so it will be renewed
+            self.close()
+            self._last_response = None
```
Comment (maroun-stratoscale) [inline hammock/templates/client/file.j2]: Why do you have to set it to `None` here?

File: hammock/templates/client/file.j2 (hunk 1 above)
Comment (shoham-stratoscale) [inline hammock/templates/client/file.j2]: :+1:

File: hammock/templates/client/file.j2 (hunk 2 above)
Comment (shoham-stratoscale) [inline hammock/templates/client/file.j2]: It's not a must. It's just so in this method context, the next time it's called, it won't do anything until someone sets the _last_reponse_time

**Comment by shimshon-stratoscale** [comment]
@shoham-stratoscale - now everyone has to update hammock versions to generate updated clients, and then everyone has to update clients, right? (obviously we should start where it hurts the most)

**Comment by adir-stratoscale** [comment]
@shoham-stratoscale isn't it a feature of requests that we can use?

**Comment by shoham-stratoscale** [comment]
@shimshon-stratoscale yes
And also change their uwsgi yaml files like [here](https://github.com/Stratoscale/melet-api/pull/25)

**Comment by shoham-stratoscale** [comment]
@adir-stratoscale Unfortunately no. looked with @eyal-stratoscale

**Comment by shoham-stratoscale** [comment]
@eyal-stratoscale pointed out some problems with this implementation.
I'm going to change to the retry-method, and open a new PR

**Pull request closed without merging**